File: python/파이썬 스터디 문제 풀이/정렬/가장 큰 수.py
from functools import cmp_to_key

def solution(numbers):
    numbers = [str(number) for number in numbers]
    numbers = sorted(numbers, reverse = True, key = cmp_to_key(comparator))
    return str(int(''.join(numbers)))

def comparator(a,b):
    c1 = int(a+b)
    c2 = int(b+a)
    return (c1> c2) - (c2>c1)

File: JavaScript/정렬/가장 큰 수.js
function solution(numbers) {
  numbers.sort((a, b) => {
    const first = String(a) + String(b);
    const second = String(b) + String(a);

    return Number(second) - Number(first);
  });

  return Number(numbers.join("")) ? numbers.join("") : "0";
}

console.log(solution([3, 30, 34, 5, 9]));
